package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -45,37 +73,287 @@ define([
      * @description Creates a base `Model`.
      * @param {pentaho.visual.base.spec.IModel} modelSpec A plain object containing the model specification.
      */
-    var Model = Complex.extend({
-          meta: {
-            id: "pentaho/visual/base",
-            view: "View",
-            isAbstract: true,
-            props: [
-              {
-                name: "width",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "height",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "isInteractive",
-                type: "boolean",
-                value: true
-              },
-              {
-                name: "data",
-                type: "object",
-                isRequired: true
-              }
-            ]
+    var Model = Complex.extend().implement(EventSource).extend(/** @lends pentaho.visual.base.Model# */{
+        constructor: function(modelSpec) {
+          this.base(modelSpec);
+        },
+
+        /**
+         * Modifies the current selection based on an input filter and on a selection mode.
+         *
+         * This action is the entry point for user-driven modifications of the current selection,
+         * e.g. if the user clicked a bar in a bar chart,
+         * or drew a rectangle over a set of bars in a bar chart.
+         *
+         * The event ["will:select"]{@link pentaho.visual.base.events.WillSelect}
+         * is first emitted and processed.
+         * Its event listeners can be attributed a _priority_,
+         * and can be regarded as operations in a processing pipeline that are allowed to:
+         * - cancel the event,
+         * - mutate the input filter
+         * - mutate the selection mode.
+         *
+         * Afterwards, [_changeSelection]{@link pentaho.visual.base.Model#_changeSelection}
```

Comment:
The current selection filter. Also, not sure it is relevant to mention the protected method in this context, where we're interested in explaining the flow, conceptually. You can have a "For implementors" section explaining the methods involved, what each is for, which you would override. Or, simply, rely on the `@see` tag. Other, non-deriving users, don't care about those details.